tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+                        self.cco.setDetectorPos(pupilPos=pupilPos, detectorPos=detectorPos, detector=detector,
+                                                beam=beam)
+                        beamInfo = self.cco[beam]
+                        self.assertTrue(beamInfo.isOnDetector)
+                        self.assertEqual(beamInfo.detectorName, desiredDetector)
+                        self.assertPairsAlmostEqual(beamInfo.pupilPos, pupilPos,
+                                                    maxDiff=self.maxPupilPosErr)
+                        self.assertPairsAlmostEqual(beamInfo.detectorPos, detectorPos,
+                                                    maxDiff=self.maxDetectorPosErr)
+                        self.checkOrientation()
+
+    def testSetFocalPlanePos(self):
+        maxErrMm = self.scc.pixelSizeMm * 0.0001
+        for focalPlanePos, pupilPos, beam in itertools.product(
+            ((0, 0), (500, 200), (25, 850)),
+            ((0, 0), (0, 5000), (-5000, 0), (5000, -5000)),
+            self.cco.beamNames,
+        ):
+            with self.subTest(focalPlanePos=focalPlanePos, pupilPos=pupilPos, beam=beam):
+                self.cco.setFocalPlanePos(pupilPos=pupilPos, focalPlanePos=focalPlanePos, beam=beam)
+                beamInfo = self.cco[beam]
+                self.assertTrue(beamInfo.isOnFocalPlane)
+                errMm = math.hypot(*np.subtract(beamInfo.focalPlanePos, focalPlanePos))
+                self.assertLess(errMm, maxErrMm)
+                self.assertPairsAlmostEqual(beamInfo.pupilPos, pupilPos, self.maxPupilPosErr)
+
+    def testSetPupilFieldAngleZero(self):
+        """Test setPupilFieldAngle for zero field angle and various points on the pupil
+        """
+        for pupilPos in ((0, 5000), (-5000, 0), (5000, -5000)):
+            with self.subTest(pupilPos=pupilPos):
+                self.cco.setPupilFieldAngle(pupilPos=pupilPos)
+
+                # the telescope should be pointing in the opposite direction of the CBP
+                telDir = self.cco.telAzAltInternal.getVector()
+                cbpDir = self.cco.cbpAzAltInternal.getVector()
+                negativeCbpDir = -np.array(cbpDir, dtype=float)
+                np.testing.assert_allclose(telDir, negativeCbpDir, atol=1e-15)
+
+                maxDetectorPosError = 0.0001
+
+                # beam 0 should be pointed to the center of the pupil, normal to the pupil
+                # and land on the center of the focal plane and the center of detector D0
+                beamInfo0 = self.cco["0"]
+                self.assertEqual(beamInfo0.name, "0")
+                self.assertPairsAlmostEqual(beamInfo0.holePos, (0, 0))
+                self.assertTrue(beamInfo0.isOnPupil)
+                self.assertTrue(beamInfo0.isOnFocalPlane)
+                self.assertTrue(beamInfo0.isOnDetector)
+                self.assertPairsAlmostEqual(beamInfo0.focalPlanePos, (0, 0), maxDiff=maxDetectorPosError)
```

Comment:
Shouldn't that be `maxFocalPlanePosErr` instead of `maxDetectorPosError`?

Reply:
Good catch! Fixed in two places.